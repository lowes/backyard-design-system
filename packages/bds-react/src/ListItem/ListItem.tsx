import * as React from 'react'
import { ListItemWrapper } from './styles/ListItemWrapper'
import type { BackyardBaseProps } from '../utils/typings/BackyardProps'
import type { BDSForwardRef } from '../utils/typings/BDSComponentProps'

const ListItem: BDSForwardRef<
    ListItemProps & {
        ref?: any
    }
> = React.forwardRef<ListItemRef, ListItemProps>(function ListItem({ children, ...props }, ref) {
    return (
        <ListItemWrapper ref={ref} {...props}>
            {children}
        </ListItemWrapper>
    )
})

type ListItemRef = HTMLLIElement

interface ListItemProps extends BackyardBaseProps<ListItemRef> {}

ListItem.bdsName = 'ListItem'

export { ListItem }

export type { ListItemRef, ListItemProps }

export default ListItem
