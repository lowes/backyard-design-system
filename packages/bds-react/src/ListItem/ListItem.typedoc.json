{
	"id": 0,
	"name": "@lowes-tech/bds-react/ListItem",
	"kind": 1,
	"kindString": "Project",
	"flags": {},
	"children": [
		{
			"id": 7,
			"name": "default",
			"kind": 16777216,
			"kindString": "Reference",
			"flags": {},
			"target": 1
		},
		{
			"id": 5,
			"name": "ListItemProps",
			"kind": 256,
			"kindString": "Interface",
			"flags": {},
			"children": [
				{
					"id": 6,
					"name": "as",
					"kind": 1024,
					"kindString": "Property",
					"flags": {
						"isOptional": true
					},
					"type": {
						"type": "reference",
						"typeArguments": [
							{
								"type": "intrinsic",
								"name": "any"
							}
						],
						"name": "ElementType"
					},
					"inheritedFrom": {
						"type": "reference",
						"name": "BackyardBaseProps.as"
					}
				}
			],
			"groups": [
				{
					"title": "Properties",
					"kind": 1024,
					"children": [
						6
					]
				}
			],
			"extendedTypes": [
				{
					"type": "reference",
					"typeArguments": [
						{
							"type": "reference",
							"id": 4,
							"name": "ListItemRef"
						}
					],
					"name": "BackyardBaseProps"
				}
			]
		},
		{
			"id": 4,
			"name": "ListItemRef",
			"kind": 4194304,
			"kindString": "Type alias",
			"flags": {},
			"type": {
				"type": "reference",
				"name": "HTMLLIElement"
			}
		},
		{
			"id": 1,
			"name": "ListItem",
			"kind": 32,
			"kindString": "Variable",
			"flags": {},
			"type": {
				"type": "reference",
				"typeArguments": [
					{
						"type": "intersection",
						"types": [
							{
								"type": "reference",
								"id": 5,
								"name": "ListItemProps"
							},
							{
								"type": "reflection",
								"declaration": {
									"id": 2,
									"name": "__type",
									"kind": 65536,
									"kindString": "Type literal",
									"flags": {},
									"children": [
										{
											"id": 3,
											"name": "ref",
											"kind": 1024,
											"kindString": "Property",
											"flags": {
												"isOptional": true
											},
											"type": {
												"type": "intrinsic",
												"name": "any"
											}
										}
									],
									"groups": [
										{
											"title": "Properties",
											"kind": 1024,
											"children": [
												3
											]
										}
									]
								}
							}
						]
					}
				],
				"name": "BDSForwardRef"
			},
			"defaultValue": "..."
		}
	],
	"groups": [
		{
			"title": "References",
			"kind": 16777216,
			"children": [
				7
			]
		},
		{
			"title": "Interfaces",
			"kind": 256,
			"children": [
				5
			]
		},
		{
			"title": "Type aliases",
			"kind": 4194304,
			"children": [
				4
			]
		},
		{
			"title": "Variables",
			"kind": 32,
			"children": [
				1
			]
		}
	]
}