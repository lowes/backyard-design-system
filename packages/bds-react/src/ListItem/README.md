# @lowes-tech/bds-react/ListItem

## Interfaces

- [ListItemProps](interfaces/ListItemProps.md)

## References

### default

Renames and re-exports [ListItem](README.md#listitem)

## Type aliases

### ListItemRef

Ƭ **ListItemRef**: `HTMLLIElement`

## Variables

### ListItem

• **ListItem**: `BDSForwardRef`<[`ListItemProps`](interfaces/ListItemProps.md) & { `ref?`: `any`  }\>
