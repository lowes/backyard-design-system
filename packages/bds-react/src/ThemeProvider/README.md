# @lowes-tech/bds-react/ThemeProvider

## Namespaces

- [ThemeProvider](modules/ThemeProvider.md)

## Interfaces

- [BackyardTheme](interfaces/BackyardTheme.md)
- [ThemeProviderProps](interfaces/ThemeProviderProps.md)

## References

### default

Renames and re-exports [ThemeProvider](README.md#themeprovider)

## Type aliases

### BaseTheme

Ƭ **BaseTheme**: typeof `Theme`

## Variables

### ThemeConsumer

• **ThemeConsumer**: `React.Consumer`<[`BackyardTheme`](interfaces/BackyardTheme.md)\> = `StyledComponentsThemeConsumer`

ThemeConsumer same as 'styled-components'

___

### ThemeContext

• **ThemeContext**: `React.Context`<[`BackyardTheme`](interfaces/BackyardTheme.md)\> = `StyledComponentsThemeContext`

ThemeContext same as 'styled-components'

___

### ThemeProvider

• **ThemeProvider**: `BDSFunctional`<[`ThemeProviderProps`](interfaces/ThemeProviderProps.md)\>

Backyard React Theme Provider

 <ThemeProvider theme="light" font="fellix" someOtherContext="value">
     <Grid.Container>
         ...
             <Button />
         ...
     </Grid.Container>
 </ThemeProvider>

Custom Theme Provider wrapper to provide extra context and overrides for Backyard Themes.

## Functions

### dropShadow

▸ `Const` **dropShadow**(`shadow`, `defaultShadow?`): `string`

Converts box-shadow to drop-shadow, if necessary.

Use for `filter` css instead of `box-shadow`.

**`example`**
```
dropShadow(theme.shadows.shadow_04)
 => 'drop-shadow(0px 2px 1px rgba(0, 0, 0, 0.2)) 
         drop-shadow(0px 1px 1px rgba(0, 0, 0, 0.14)) 
         drop-shadow(0px 1px 3px rgba(0, 0, 0, 0.12))'
```

#### Parameters

| Name | Type | Default value | Description |
| :------ | :------ | :------ | :------ |
| `shadow` | `string` | `undefined` | box-shadow |
| `defaultShadow` | `string` | `'none'` | default shadow if none given |

#### Returns

`string`

___

### getShape

▸ `Const` **getShape**(`shapeProp`, `shapeContext`): ``"rounded"`` \| ``"squared"``

Get the shape or shape context in priority

#### Parameters

| Name | Type | Description |
| :------ | :------ | :------ |
| `shapeProp` | `ExtendedShapes` | shape value from props |
| `shapeContext` | ``"rounded"`` \| ``"squared"`` | shape value from context |

#### Returns

``"rounded"`` \| ``"squared"``

___

### splitShadows

▸ `Const` **splitShadows**(`shadows`, `defaultShadow`): `string`[]

Split the shadows in a given string

**`example`**
```
splitShadows('0px 2px 1px rgba(0, 0, 0, 0.2), 0px 1px 1px rgba(0, 0, 0, 0.14), 0px 1px 3px rgba(0, 0, 0, 0.12)')
 => [
     '0px 2px 1px rgba(0, 0, 0, 0.2)',
     '0px 1px 1px rgba(0, 0, 0, 0.14)',
     '0px 1px 3px rgba(0, 0, 0, 0.12)'
 ]
```

#### Parameters

| Name | Type | Description |
| :------ | :------ | :------ |
| `shadows` | `string` | shadow string from backyard |
| `defaultShadow` | `string` | default shadow if none given |

#### Returns

`string`[]

___

### useBackyardTheme

▸ `Const` **useBackyardTheme**(`props?`): [`BackyardTheme`](interfaces/BackyardTheme.md)

#### Parameters

| Name | Type |
| :------ | :------ |
| `props` | `UseBackyardThemeProps` |

#### Returns

[`BackyardTheme`](interfaces/BackyardTheme.md)

___

### validateBackyardTheme

▸ `Const` **validateBackyardTheme**(`theme`, `name`): `boolean`

#### Parameters

| Name | Type |
| :------ | :------ |
| `theme` | [`BackyardTheme`](interfaces/BackyardTheme.md) |
| `name` | `string` |

#### Returns

`boolean`
