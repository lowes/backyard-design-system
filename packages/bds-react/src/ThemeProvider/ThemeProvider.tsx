import * as React from 'react'
import {
    ThemeProvider as StyledComponentsThemeProvider,
    ThemeContext as StyledComponentsThemeContext,
    ThemeConsumer as StyledComponentsThemeConsumer,
} from 'styled-components'

import light from '@lowes-tech/bds-tokens/v3/light/_theme'
import dark from '@lowes-tech/bds-tokens/v3/dark/_theme'
import type Theme from '@lowes-tech/bds-tokens/v3/light/_theme'

import useMediaQuery from '../utils/hooks/useMediaQuery'

import { BDSFunctional } from '../utils/typings/BDSComponentProps'

type BaseTheme = typeof Theme

/**
 * Combine themes into centralized object for easy access
 *
 * @todo contrast needs to be set here when tokens are created - currently defaults to light theme
 */
const themes = {
    light,
    dark,
}

/**
 * ThemeContext same as 'styled-components'
 */
const ThemeContext: React.Context<BackyardTheme> = StyledComponentsThemeContext

/**
 * ThemeConsumer same as 'styled-components'
 */
const ThemeConsumer: React.Consumer<BackyardTheme> = StyledComponentsThemeConsumer

/**
 * Backyard React Theme Provider
 *
 *  <ThemeProvider theme="light" font="roboto" someOtherContext="value">
 *      <Grid.Container>
 *          ...
 *              <Button />
 *          ...
 *      </Grid.Container>
 *  </ThemeProvider>
 *
 * Custom Theme Provider wrapper to provide extra context and overrides for Backyard Themes.
 */

const ThemeProvider: BDSFunctional<ThemeProviderProps> = ({
    children,
    theme: themeProp = 'light',
    override = {},
    ...context
}) => {
    // Save state of current theme and allow theme to be changed
    const [curTheme, setTheme] = React.useState(themeProp)
    // Save state of overrides and allow overrides to be changed
    const [curOverride, setOverride] = React.useState(override)
    // Save state of context and allow context to be changed
    // eslint-disable-next-line @typescript-eslint/no-use-before-define
    const [curContext, setContextState] = React.useState<BackyardThemeContext>(getContext(context))

    // Side effects to change theme
    React.useEffect(() => setTheme(themeProp), [themeProp])
    // Get current theme
    const theme = themes[curTheme] as BaseTheme

    // Default Breakpoint Schema
    const breakpoints = {
        // Extra Small
        xs: theme.grid.breakpoint.xs.max,
        xs_min: theme.grid.breakpoint.xs.min,
        xs_max: theme.grid.breakpoint.xs.max,
        // Small
        sm: theme.grid.breakpoint.sm.max,
        sm_min: theme.grid.breakpoint.sm.min,
        sm_max: theme.grid.breakpoint.sm.max,
        // Medium
        md: theme.grid.breakpoint.md.max,
        md_min: theme.grid.breakpoint.md.min,
        md_max: theme.grid.breakpoint.md.max,
        // Large
        lg: theme.grid.breakpoint.lg.max,
        lg_min: theme.grid.breakpoint.lg.min,
        lg_max: theme.grid.breakpoint.lg.max,
        // Extra Large
        xl: theme.grid.breakpoint.xl.max,
        xl_min: theme.grid.breakpoint.xl.min,
        xl_max: theme.grid.breakpoint.xl.max,
        // Extra-Extra Large
        xxl: theme.grid.breakpoint.xxl.max,
        xxl_min: theme.grid.breakpoint.xxl.min,
        xxl_max: theme.grid.breakpoint.xxl.max,
    }

    // Media Query Flags
    const mediaQueries = {
        // Device Queries
        isMobile: useMediaQuery(`only screen and (max-width: ${breakpoints.md_min})`),
        isTablet: useMediaQuery(`only screen and 
                 (min-width: ${breakpoints.md_min}) and 
                 (max-width: ${breakpoints.lg_max})`),
        isDesktop: useMediaQuery(`only screen and (min-width: ${breakpoints.xl_min})`),
        // Breakpoint Queries
        isXS: useMediaQuery(`only screen and 
                 (min-width: ${breakpoints.xs_min}) and 
                 (max-width: ${breakpoints.xs_max})`),
        isSM: useMediaQuery(`only screen and 
                 (min-width: ${breakpoints.sm_min}) and 
                 (max-width: ${breakpoints.sm_max})`),
        isMD: useMediaQuery(`only screen and 
                 (min-width: ${breakpoints.md_min}) and 
                 (max-width: ${breakpoints.md_max})`),
        isLG: useMediaQuery(`only screen and 
                 (min-width: ${breakpoints.lg_min}) and 
                 (max-width: ${breakpoints.lg_max})`),
        isXL: useMediaQuery(`only screen and 
                 (min-width: ${breakpoints.xl_min}) and 
                 (max-width: ${breakpoints.xl_max})`),
        isXXL: useMediaQuery(`only screen and 
                 (min-width: ${breakpoints.xxl_min})`),
    }

    /**
     * Merges `newContext` with `curContext` to allow for new context to be added
     *
     * @param {object<string, string>} newContext - New Context to be added
     */
    function getContext(newContext, oldContext = {}): BackyardThemeContext {
        const {
            font = 'roboto',
            heading,
            paragraph,
            article,
            caption,
            link,
            canvas = 'surface_default',
            shape = 'rounded',
            typography = {
                h1: {
                    desktop: {
                        font_size: light.font.tag.h1.desktop.font_size,
                        line_height: light.font.tag.h1.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.h1.mobile.font_size,
                        line_height: light.font.tag.h1.mobile.line_height,
                    },
                },
                h2: {
                    desktop: {
                        font_size: light.font.tag.h2.desktop.font_size,
                        line_height: light.font.tag.h2.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.h2.mobile.font_size,
                        line_height: light.font.tag.h2.mobile.line_height,
                    },
                },
                h3: {
                    desktop: {
                        font_size: light.font.tag.h3.desktop.font_size,
                        line_height: light.font.tag.h3.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.h3.mobile.font_size,
                        line_height: light.font.tag.h3.mobile.line_height,
                    },
                },
                h4: {
                    desktop: {
                        font_size: light.font.tag.h4.desktop.font_size,
                        line_height: light.font.tag.h4.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.h4.mobile.font_size,
                        line_height: light.font.tag.h4.mobile.line_height,
                    },
                },
                h5: {
                    desktop: {
                        font_size: light.font.tag.h5.desktop.font_size,
                        line_height: light.font.tag.h5.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.h5.mobile.font_size,
                        line_height: light.font.tag.h5.mobile.line_height,
                    },
                },
                h6: {
                    desktop: {
                        font_size: light.font.tag.h6.desktop.font_size,
                        line_height: light.font.tag.h6.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.h6.mobile.font_size,
                        line_height: light.font.tag.h6.mobile.line_height,
                    },
                },
                body_1: {
                    desktop: {
                        font_size: light.font.tag.body_1.desktop.font_size,
                        line_height: light.font.tag.body_1.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.body_1.mobile.font_size,
                        line_height: light.font.tag.body_1.mobile.line_height,
                    },
                },
                body_2: {
                    desktop: {
                        font_size: light.font.tag.body_2.desktop.font_size,
                        line_height: light.font.tag.body_2.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.body_2.mobile.font_size,
                        line_height: light.font.tag.body_2.mobile.line_height,
                    },
                },
                article: {
                    desktop: {
                        font_size: light.font.tag.article.desktop.font_size,
                        line_height: light.font.tag.article.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.article.mobile.font_size,
                        line_height: light.font.tag.article.mobile.line_height,
                    },
                },
                caption: {
                    desktop: {
                        font_size: light.font.tag.caption.desktop.font_size,
                        line_height: light.font.tag.caption.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.caption.mobile.font_size,
                        line_height: light.font.tag.caption.mobile.line_height,
                    },
                },
                footnote: {
                    desktop: {
                        font_size: light.font.tag.footnote.desktop.font_size,
                        line_height: light.font.tag.footnote.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.footnote.mobile.font_size,
                        line_height: light.font.tag.footnote.mobile.line_height,
                    },
                },
                label: {
                    desktop: {
                        font_size: light.font.tag.label.desktop.font_size,
                        line_height: light.font.tag.label.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.label.mobile.font_size,
                        line_height: light.font.tag.label.mobile.line_height,
                    },
                },
                overline: {
                    desktop: {
                        font_size: light.font.tag.overline.desktop.font_size,
                        line_height: light.font.tag.overline.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.overline.mobile.font_size,
                        line_height: light.font.tag.overline.mobile.line_height,
                    },
                },
                helper_text: {
                    desktop: {
                        font_size: light.font.tag.helper_text.desktop.font_size,
                        line_height: light.font.tag.helper_text.desktop.line_height,
                    },
                    mobile: {
                        font_size: light.font.tag.helper_text.mobile.font_size,
                        line_height: light.font.tag.helper_text.mobile.line_height,
                    },
                },
            },
        } = newContext

        return {
            ...oldContext,
            font,
            heading: heading || font,
            paragraph: paragraph || font,
            article: article || font,
            caption: caption || font,
            link: link || font,
            canvas,
            shape,
            typography,
            ...newContext,
        }
    }

    /**
     * Wrap `setContext` to modify `newContext` before it is set into `curContext`
     *
     * @param {object<string, string>} newContext
     */
    function setContext(newContext: BackyardThemeContext) {
        setContextState((oldContext) => getContext(newContext, oldContext))
    }

    // Set backyard theme
    const backyardTheme: BackyardTheme = {
        ...(theme as BaseTheme),
        context: curContext,
        breakpoints,
        ...mediaQueries,
        setTheme,
        setOverride,
        setContext,
        ...curOverride,
    }

    return (
        <StyledComponentsThemeProvider theme={backyardTheme}>
            {children}
        </StyledComponentsThemeProvider>
    )
}

interface BackyardThemeContext {
    /**
     * Default font contexts
     */
    font: string
    heading: string
    paragraph: string
    article: string
    caption: string
    link: string
    /**
     * Default canvas color
     */
    canvas: keyof BackyardTheme['color']
    /**
     * Default shape context
     */
    shape: 'rounded' | 'squared'
    // Allow any other values to be added to context
    // Teams will need to extend `BackyardThemeContext` with their
    // own types for TypeScript to provide assistance with them
    [key: string]: any
}

interface BackyardTheme extends BaseTheme {
    /**
     * Context can be any object
     */
    context: BackyardThemeContext
    /**
     * Whether or not current viewport is desktop, tablet, or mobile
     */
    isDesktop: boolean
    isTablet: boolean
    isMobile: boolean
    /**
     * Record of breakpoints provided to application
     */
    breakpoints: Record<string, string>
    /**
     * `setTheme` dispatch to change the theme state of the the Backyard theme
     */
    setTheme: React.Dispatch<React.SetStateAction<'light' | 'dark'>>
    /**
     * `setOverride` dispatch to change the overrides of the Backyard theme
     */
    setOverride: React.Dispatch<React.SetStateAction<object>>
    /**
     * `setContext` dispatch to change the context of the Backyard theme
     * Can store anything such as the font, heading, etc.
     */
    setContext: (newContext: any) => void
    /**
     * Name of the current theme
     */
    name: 'light' | 'dark' | string
}

interface ThemeProviderProps {
    /**
     * React Children
     */
    children: React.ReactNode
    /**
     * Selected Theme
     */
    theme?: 'light' | 'dark'
    /**
     * Default shape of all components
     */
    shape?: 'rounded' | 'squared'
    /**
     * Canvas color token
     */
    canvas?: keyof BackyardTheme['color']
    /**
     * Overrides to override Backyard theme with
     */
    override?: object
    /**
     * Extra Context to be made available to child 'styled-components'
     * ex. font
     */
    context?: object
}

ThemeProvider.bdsName = 'ThemeProvider'

export { ThemeProvider, ThemeContext, ThemeConsumer }

export type { BaseTheme, BackyardTheme, ThemeProviderProps }

export default ThemeProvider
