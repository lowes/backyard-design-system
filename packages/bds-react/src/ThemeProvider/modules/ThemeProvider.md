# Namespace: ThemeProvider

## Variables

### bdsName

â€¢ **bdsName**: `string`
