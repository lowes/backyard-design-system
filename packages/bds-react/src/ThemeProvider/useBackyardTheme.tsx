import * as React from 'react'

import { ThemeContext } from './ThemeProvider'
import type { BackyardTheme } from './ThemeProvider'

const validateBackyardTheme = (theme: BackyardTheme, name: BackyardTheme['name']) => {
    if (!theme?.name) {
        // eslint-disable-next-line no-console
        throw Error(
            `Backyard: A Backyard component ${
                name ? `(\`${name}\`)` : ''
            } is missing a Backyard Theme.\n` +
                'This component uses Backyard tokens and a Backyard Theme is required.\n' +
                `Please wrap ${
                    name ? `\`${name}\` or` : ''
                } your entire application in a Backyard \`ThemeProvider\`.\n` +
                'For a getting started guide, see \n' +
                '`https://backyard.lowes.com/GetStarted/Develop#create-theme-provider`',
        )
    }

    return true
}

const useBackyardTheme = (props: UseBackyardThemeProps = {}) => {
    const { validate = false, name }: UseBackyardThemeProps = props

    const theme = React.useContext<BackyardTheme>(ThemeContext)

    if (validate && process.env.NODE_ENV !== 'production') {
        React.useEffect(() => {
            validateBackyardTheme(theme, name)
        }, [theme])
    }

    return theme
}

interface UseBackyardThemeProps {
    /**
     * Whether or not to validate component
     */
    validate?: boolean
    /**
     * Name of the component to validate
     */
    name?: string
}

export { useBackyardTheme, validateBackyardTheme }

export default useBackyardTheme
