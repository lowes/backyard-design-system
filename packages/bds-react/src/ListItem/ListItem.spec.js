import * as React from 'react'
import { render } from '../../test-utils'
import { ThemeProvider } from '../ThemeProvider'
import ListItem from '../ListItem'

describe('ListItem Snapshots', () => {
    test('ListItem default snapshot', () => {
        const { asFragment } = render(
            <ListItem>List item</ListItem>
        )

        expect(asFragment()).toMatchSnapshot()
    })

    describe('themes', () => {
        const themes = ['light', 'dark']

        themes.forEach((theme) => {
            test(`ListItem ${theme} theme snapshot`, () => {
                const { asFragment } = render(
                    <ThemeProvider theme={theme}>
                        <React.Fragment>
                            <ListItem>Item one</ListItem>
                        </React.Fragment>
                    </ThemeProvider>
                )
                expect(asFragment()).toMatchSnapshot()
            })
        })
    })
})