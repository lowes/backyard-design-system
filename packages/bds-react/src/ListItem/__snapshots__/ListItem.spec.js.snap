// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Snapshots ListItem default snapshot 1`] = `
<DocumentFragment>
  .c0 {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-20);
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

@media only screen and (min-width:82rem) {
  .c0 {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }
}

<li
    class="c0"
  >
    List item
  </li>
</DocumentFragment>
`;

exports[`ListItem Snapshots themes ListItem dark theme snapshot 1`] = `
<DocumentFragment>
  .c0 {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-20);
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

@media only screen and (min-width:82rem) {
  .c0 {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }
}

<li
    class="c0"
  >
    Item one
  </li>
</DocumentFragment>
`;

exports[`ListItem Snapshots themes ListItem light theme snapshot 1`] = `
<DocumentFragment>
  .c0 {
  font-size: var(--bds-sizes-size-14);
  line-height: var(--bds-sizes-size-20);
  font-family: Roboto,Helvetica,Arial,sans-serif;
  font-weight: 500;
}

@media only screen and (min-width:82rem) {
  .c0 {
    font-size: var(--bds-sizes-size-16);
    line-height: var(--bds-sizes-size-24);
  }
}

<li
    class="c0"
  >
    Item one
  </li>
</DocumentFragment>
`;
